src/main/scala/bot/domain/states/AdviseState.scala

```diff
@@ -8,12 +8,12 @@ case class AdviseState(
   wineType: Option[String],
   priceMin: Option[Int],
   priceMax: Option[Int],
-) {
+) extends State {
   def getCountryCode: Option[CountryCode] = this.country.flatMap(CountryButton.withNameOption(_).map(_.countryCode))
```

Comment:
Instead of doing `this.country`, either call `country` directly
Or do it like this, if the value name is shadowed:
```
case class Foo(bar: Bar) { self =>
  def addBar(bar: Bar): Foo = copy(bar = self.bar + bar)
}
```